mocks/mock_domains.go

```diff
@@ -140,6 +142,45 @@ func (ms *Server) listDomains(w http.ResponseWriter, r *http.Request) {
 	})
 }
 
+func (ms *Server) listIPDomains(w http.ResponseWriter, r *http.Request) {
+	defer ms.mutex.Unlock()
+	ms.mutex.Lock()
+
+	var list []mtypes.Domain
+	for _, domain := range ms.domainList {
+		list = append(list, domain.Domain)
+	}
+
+	skip := stringToInt(r.FormValue("skip"))
+	limit := stringToInt(r.FormValue("limit"))
+	if limit == 0 {
+		limit = 100
+	}
+
+	if skip > len(list) {
+		skip = len(list)
+	}
+
+	end := limit + skip
+	if end > len(list) {
+		end = len(list)
+	}
+
+	// If we are at the end of the list
+	if skip == end {
+		toJSON(w, mtypes.ListDomainsResponse{
+			TotalCount: len(list),
+			Items:      []mtypes.Domain{},
+		})
```

Comment:
That's not what that endpoint returns:
<img width="520" alt="image" src="https://github.com/user-attachments/assets/0f1ad6fc-6319-4c34-bf74-a00ebbd50e10" />
https://documentation.mailgun.com/docs/mailgun/api-reference/openapi-final/tag/IPs/#tag/IPs/operation/GET-v3-ips--ip--domains